### PR TITLE
fix: correct installed_pkgs query. prompt when unattached

### PR DESCRIPTION
**Pull request by blackboxsw**
Prompt for ua attach when unattached and `ua fix CVE-XX` reports
an available package affected.

Add a util.prompt_choices function which defines the input string
and desirable choices/responses. prompt_choices will loop until the
valid choice is provided.

In testing ua fix CVE-2021-27135 it was noted that xterm wasn't
listed as installed. Fix dpkg-query to determine package name based
on either ${Source} or ${Package} if Source is not set.

Fixes: #1397, #1398

## Proposed Commit Message
Rebase and merge

## Test Steps
```bash
cat > 1.yaml <<EOF
#cloud-config                                                                   
package_upgrade: true                                                           
apt:                                                                            
  sources:                                                                      
    ua.list:                                                                    
       source: deb http://ppa.launchpad.net/canonical-server/ua-client-daily/ubuntu $RELEASE main
       keyid: 94E187AD53A59D1847E4880F8A295C4FB8B190B7 
packages: [ubuntu-advantage-tools]
EOF

tox -e lxc launch ubuntu-daily:bionic dev-b -c user.user-data="$(cat 1.yaml)"
lxc file push uaclient/security.py dev-b/usr/lib/python3/dist-packages/uaclient/
lxc file push uaclient/status.py dev-b/usr/lib/python3/dist-packages/uaclient/
lxc file push uaclient/util.py dev-b/usr/lib/python3/dist-packages/uaclient/

# downgrade xterm
lxc exec dev-b apt-get install xterm=330-1ubuntu2
lxc exec dev-b ua fix CVE-2021-27135
```
## Desired commit type::

 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [x] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist:

 - [x] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly # maybe will add them after initial review
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

File: uaclient/security.py [hunk 1]
```diff
@@ -410,6 +420,8 @@ def prompt_for_affected_packages(
     )
     msg_suffix = ": " + ", ".join(affected_pkg_status.keys())
     msgs = [msg + msg_suffix]
+    upgrade_packages = []
+    cloud_type = get_cloud_type()
```
Comment (lucasmoura) [inline uaclient/security.py:424]: It seems that we are not using this variable ? Also, why would we need the cloud_type in this function ?

File: uaclient/tests/test_security.py [hunk 2]
```diff
@@ -543,8 +552,81 @@ def test_result_keyed_by_source_package_name(
...
+                {"sl": "2.0"},
+                [
+                    "1 affected package is installed: sl\n(1/1) sl:\n",
+                    "The update is not installed because this system is not"
+                    " attached to a\nsubscription that covers these packages.",
+                ],
+            ),
+        ),
+    )
+    @mock.patch("uaclient.security.get_cloud_type", return_value="aws")
```
Comment (lucasmoura) [inline uaclient/tests/test_security.py:610]: If we remove the `get_cloud_type` call, we can drop this mock

**Comment by lucasmoura** [comment]
@blackboxsw Another question, when reproducing the test scenario you proposed we are downgrading a package which already has a fix on `Ubuntu Standard Updates`. However, we trigger the user in that scenario that it needs an uaclient subscription to solve that issue, which is not true.

Shouldn't we just ask that question if the updates cannot be found on the existing ubuntu repositories ?

File: uaclient/security.py (hunk 1 above)
Comment (blackboxsw) [reply uaclient/security.py:424]: Ahh this should be a followup PR. I accidentally left this in, it will be used later. I'll back it out of this PR

**Comment by blackboxsw** [comment]
> @blackboxsw Another question, when reproducing the test scenario you proposed we are downgrading a package which already has a fix on `Ubuntu Standard Updates`. However, we trigger the user in that scenario that it needs an uaclient subscription to solve that issue, which is not true.
> 
> Shouldn't we just ask that question if the updates cannot be found on the existing ubuntu repositories ?

Thanks, I originally thought to wait on this, but it was low-hanging fruit for this branch. I refactored functions and made this part smarter so we don't have to refactor later.

**Review by lucasmoura — APPROVED** [review]

**Pull request merged**